src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,104 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.ValidationFailedException;
+import core.basesyntax.model.User;
+import java.util.Arrays;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User(2L, "my_login", "123456", 25);
+        Storage.people.addAll(Arrays.asList(new User(5L, "petro34", "123679", 21),
+                new User(4L, "oleksandr9", "5328409", 18),
+                new User(8L, "semen56", "653hddxe", 19),
+                new User(9L, "pavlo676", "50ujgda", 22),
+                new User(7L, "my_login2", "705jpteq", 25)));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(ValidationFailedException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(ValidationFailedException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(ValidationFailedException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin("");
+        assertThrows(ValidationFailedException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        user.setPassword("");
+        assertThrows(ValidationFailedException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        user.setLogin("login");
+        assertThrows(ValidationFailedException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
```

Comment:
The test method `register_shortPassword_notOk` does not test the edge case for the password length. According to the checklist, you should test for the edge case where the password length is just one character less than the minimum required length (e.g., 5 characters if the minimum is 6).